src/lib/Scenes/Home/Components/Sales/index.tsx

```diff
@@ -1,114 +1,91 @@
+  contentContainer: {
+    justifyContent: "space-between",
+    padding: 10,
+    display: "flex",
+  },
+})
 
-  renderList(itemData) {
-    const numColumns = Dimensions.get("window").width > 700 ? 4 : 2
-    return (
-      <FlatList
-        contentContainerStyle={{ justifyContent: "space-between", padding: 5, display: "flex" }}
-        data={itemData.data}
-        numColumns={numColumns}
-        keyExtractor={(item, index) => item.__id}
-        renderItem={d => {
-          return (
-            <TouchableWithoutFeedback onPress={this.handleTap.bind(this, d)}>
-              <View style={{ marginRight: 10, marginBottom: 10 }}>
-                <SaleItem key={d.index} sale={d.item} />
-              </View>
-            </TouchableWithoutFeedback>
-          )
-        }}
-      />
-    )
+class Sales extends React.Component<Props> {
+  get data() {
+    const { sales } = this.props
+    const liveAuctions = sales.filter(a => !!a.live_start_at)
+    const timedAuctions = sales.filter(a => !a.live_start_at)
+
+    return {
+      liveAuctions,
+      timedAuctions,
+    }
   }
 
   render() {
-    const sales = this.props.sales
-    const liveAuctions = sales.filter(a => !!a.live_start_at)
-    const timedAuctions = sales.filter(a => !a.live_start_at)
     const sections = [
       {
-        data: [
-          {
-            data: liveAuctions,
-          },
-        ],
+        data: [{ data: this.data.liveAuctions }],
         title: "Current Live Auctions",
+        renderItem: itemData => <SaleList {...itemData} />,
       },
       {
-        data: [
-          {
-            data: timedAuctions,
-          },
-        ],
+        data: [{ data: this.data.timedAuctions }],
         title: "Current Timed Auctions",
+        renderItem: itemData => <SaleList {...itemData} />,
+      },
+      {
+        data: [{ data: [] }],
+        title: "Lots by Artists You Follow",
+        renderItem: () => <LotsByFollowedArtists />,
       },
     ]
 
     return (
       <SectionList
-        contentContainerStyle={{
-          justifyContent: "space-between",
-          padding: 15,
-          display: "flex",
-        }}
+        contentContainerStyle={SectionListStyles.contentContainer}
         stickySectionHeadersEnabled={false}
         sections={sections}
         keyExtractor={(item, index) => item.id}
-        renderItem={itemData => {
-          return this.renderList(itemData.item)
-        }}
-        renderSectionHeader={({ section }) =>
-          <SectionHeader>
-            <SectionTitle>
-              {section.title}
-            </SectionTitle>
-          </SectionHeader>}
       />
     )
   }
 }
 
+export function SalesRenderer({ render }) {
+  return (
+    <QueryRenderer
+      environment={createEnvironment()}
+      query={graphql`
+        query SalesRendererQuery {
+          sales: sales(live: true, is_auction: true) {
+            ...Sales_sales
+          }
+        }
```

Comment:
Similar to `<LotsByFollowedArtists />`, I moved this out of its own file and into the root so that data parent/child dependencies can be easily expressed side-by-side for clarity and co-location.

Reply:
I like it, colocating the `<QueryRenderer />` makes sense. As you mentioned, they need to be at the root.

Reply:
Am I correct in understanding that there are now multiple query renderers for the one ‘auctions’ view in the home section? That would result in multiple requests, which is not what we want, we want a single query for _all_ the initial data of the view.

Reply:
Additionally these query renderers don’t appear to share environments, which they ideally would.

Reply:
@alloy No there's still only one query renderer for the Auctions view, we just moved it to another file. Instead of a separate file now (https://github.com/artsy/emission/blob/master/src/lib/Scenes/Home/index.tsx#L11) we export from the same place as the top level Auctions component (https://github.com/artsy/emission/blob/4545e9ed2c6ed7af4df4d2a18d12cfabf35de639/src/lib/Scenes/Home/Components/Sales/index.tsx#L68)

In terms of the renderers not sharing environments, it is indeed not the case right now as we're calling `createEnvironment()` again. We should export one instance of the class from a module and use it throughout instead.

Reply:
Oh, maybe I’m just misunderstanding where what view is shown then. Are the `QueryRenderer` in `LotsByFollowedArtists` and now `Sales` not for components used in the same view?

Reply:
@l2succes Reading [this line](https://github.com/artsy/emission/pull/851/files#diff-a08fc595f15caaf7c0686f93c2fcce70R53), I’m actually pretty sure that a new query renderer and thus request is made for each `renderItem` pass?

Reply:
Ok, I just noticed what you referring to, there is indeed an instance of `QueryRenderer` being instantiated in `<LotsByFollowedArtists />` which wasn't there before and it slipped past me so my mistake.
However, I should note each section in the top level `<SectionList />` only has one row. `renderItem` only gets called once per render and so there's only one instance of `LotsByFollowedArtists` being created. We still end up with two instances QueryRenderer for the Auctions tab, we should only have one so the one in `LotsByFollowedArtists` needs to be removed.

@damassi Can we address this in the second PR?

Reply:
Ah I see, thanks that’s helpful information in general for me 👍

Reply:
Will do!